pkg/ipamd/datastore/data_store.go

```diff
@@ -978,8 +978,8 @@ func (ds *DataStore) GetENINeedsIP(maxIPperENI int, skipPrimary bool) *ENI {
 	ds.lock.Lock()
 	defer ds.lock.Unlock()
 	for _, eni := range ds.eniPool {
-		if skipPrimary && eni.IsPrimary {
-			ds.log.Debugf("Skip the primary ENI for need IP check")
+		if (skipPrimary && eni.IsPrimary) || eni.IsTrunk || eni.IsEFA {
```

Comment:
You need to remove `IsEFA` check... primary ENI can be EFA enabled and still have secondary Ips/prefixes for pods.

Reply:
Why did we check `EFA` in the first place? Removing the check doesn't provide the check on non primary ENIs anymore. Is that what we want?

Reply:
The AWS VPC CNI will not allocate EFA ENIs, but it can assign IP addresses to them after they are attached. Typically, customers attaching EFA ENIs already have all IP addresses assigned, but we still cannot exclude them in this check.